calculating_n50_assemblies.py

```diff
@@ -3,60 +3,81 @@
                 switch = 'on'
+    
+    if switch == 'on':
+        n50_summary(n50_array, n50_array_log, output_file, out_dir)
 
-    if switch == 'off':
+    else:
         sys.exit("No zipped or unzipped fasta assemblies are in the directory. Exiting..")
+    
+
+def n50_calc(open_file = None, n50_array = None, n50_array_log = None):
+    contig_length_dict = {}
+    for line in open_file:
+        x = re.findall(r'^>(.+?)\s', line)
+        if len(x) > 0:
+            contig_name = x[0]
+            contig_length = 0
+        else:
+            contig_length += len(line.rstrip('\n'))
+            contig_length_dict[contig_name] = contig_length
 
-def n50_calc(open_file = None, output_file = None):
-        contig_length_dict = {}
-        for line in open_file:
-            assembly = re.findall(r'^>([A-Z]+?0\d).+',line)
-            if len(assembly) > 0:
-                assembly_name = assembly[0] 
-            x = re.findall(r'^>(.+?)\s', line)
-            if len(x) > 0:
-                contig_name = x[0]
-                contig_length = 0
-            else:
-                contig_length += len(line.rstrip('\n'))
-                contig_length_dict[contig_name] = contig_length
+    total_assembly_len = 0
+    for key in contig_length_dict:
+        total_assembly_len += contig_length_dict[key]
+    l50_len = total_assembly_len/2
+    rev_sorted_contig_length_dict = sorted(contig_length_dict.items(), key=operator.itemgetter(1),reverse=True)
+    temp_sum = 0
+    for item in rev_sorted_contig_length_dict:
+        temp_sum += item[1]
+        if temp_sum >= l50_len: 
+            n50_array.append(item[1])
+            n50_array_log.append(math.log(item[1],10))
+            break
 
-        total_assembly_len = 0
-        for key in contig_length_dict:
-            total_assembly_len += contig_length_dict[key]
-        l50_len = total_assembly_len/2
-        rev_sorted_contig_length_dict = sorted(contig_length_dict.items(), key=operator.itemgetter(1),reverse=True)
-        temp_sum = 0
-        for item in rev_sorted_contig_length_dict:
-            temp_sum += item[1]
-            if temp_sum >= l50_len: 
-                output_file.write("{}\t{}\n".format(assembly_name, item[1]))
-                break
+def n50_summary(n50_array = None, n50_array_log = None, output_file = None, out_dir = None):
+    s = pd.Series(n50_array)
+    output_file.write("Number of N50 data points is {}\n".format(s.count()))
+    output_file.write("Min and Max N50 values are {} and {}\n".format(s.min(),s.max()))
+    output_file.write("Mean of N50 distribution is {:.3f}\n".format(s.mean()))
+    output_file.write("Median of N50 distribution is {}\n".format(s.median()))
+    output_file.write("Standard deviation of N50 distribution is {:.3f}\n".format(s.std()))
+    plt.hist(n50_array_log, bins = 100, color='green')
+    plt.xlabel('Log (base 10) N50 values')
+    plt.ylabel('Counts')
+    plt.title('Histogram of E. coli N50 assembly lengths')
```

Comment:
This title assumes that the program only accepts data for E. coli which is incorrect, could you please correct this.

Reply:
Do you mean just remove E. coli from the title text? or a longer way of extracting organism name from a fasta assembly header by regular expression? e.g. 
>RSZF02000001.1 Escherichia coli strain 424660 SAMN09...... 
So something like ^>.+?\s(.+?)\s(.+?)\s